### PR TITLE
Add documentation regarding GRPC streaming use case

### PR DESCRIPTION
**Pull request by tanmayv25**
Addresses #4572.

### CODE REVIEW

File: docs/inference_protocols.md [hunk 1]
```diff
@@ -42,12 +42,27 @@ that have been proposed by the [KServe
 project](https://github.com/kserve). To fully enable all capabilities
 Triton also implements a number [HTTP/REST and GRPC
```
Comment (rmccorm4) [inline docs/inference_protocols.md:43]:
```suggestion
Triton also implements [HTTP/REST and GRPC
```

File: docs/inference_protocols.md [hunk 2]
```diff
@@ -42,12 +42,27 @@ that have been proposed by the [KServe
...
-
-The HTTP/REST and GRPC protcols provide endpoints to check server and
-model health, metadata and statistics. Additional endpoints allow
-model loading and unloading, and inferencing. See the KServe and
-extension documentation for details.
+to the KServe inference protocol. GRPC protocol also provides a
+bi-directional streaming version of the inference RPC to allow a
+sequence of inference requests/responses to be sent over a
+GRPC stream. We typically recommend using the unary version for
+inference request. The streaming version should be used only if
```
Comment (rmccorm4) [inline docs/inference_protocols.md:49]:
```suggestion
inference requests. The streaming version should be used only if the
```

File: docs/inference_protocols.md [hunk 3]
```diff
@@ -42,12 +42,27 @@ that have been proposed by the [KServe
...
+situation demands it. Some of such use cases can be:
+
+* Assume a system with mulitple Triton server instances running
```
Comment (rmccorm4) [inline docs/inference_protocols.md:52]:
```suggestion
* Assume a system with multiple Triton server instances running
```

File: docs/inference_protocols.md [hunk 4]
```diff
@@ -42,12 +42,27 @@ that have been proposed by the [KServe
...
+  behind a Load Balancer. If a sequence of inference requests is
+  needed to hit the same Triton server instance. GRPC stream
```
Comment (rmccorm4) [inline docs/inference_protocols.md:54]:
```suggestion
  needed to hit the same Triton server instance, a GRPC stream
```

File: docs/inference_protocols.md [hunk 5]
```diff
@@ -42,12 +42,27 @@ that have been proposed by the [KServe
...
+  will hold a single connection throughout the lifetime and hence
+  ensure the requests are delivered to the same Triton instance.
+* If the order of requests/responses needs to be preserved over
+  the network. GRPC stream will ensure that the server receives
```
Comment (rmccorm4) [inline docs/inference_protocols.md:58]:
```suggestion
  the network, a GRPC stream will ensure that the server receives
```

**Review by rmccorm4 — CHANGES_REQUESTED** [review]
minor typos

**Review by rmccorm4 — APPROVED** [review]

**Review by jbkyang-nvi — APPROVED** [review]

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**